src/functions/edi/inbound/__tests__/handler.success-with-ack.ts

```diff
@@ -0,0 +1,134 @@
+import { handler } from "../handler.js";
+import nock from "nock";
+import { sampleS3Event } from "../__fixtures__/events.js";
+import { sdkStreamMixin } from "@aws-sdk/util-stream-node";
+import {
+  mockBucketClient,
+  mockExecutionTracking,
+  mockGuideClient,
+  mockStashClient,
+  mockTranslateClient,
+} from "../../../../lib/testing/testHelpers.js";
+import { GetObjectCommand, PutObjectCommand } from "@stedi/sdk-client-buckets";
+import { TranslateX12ToJsonCommand } from "@stedi/sdk-client-edi-translate";
+import { Readable } from "stream";
+import fs from "node:fs";
+import {
+  GetValueCommand,
+  IncrementValueCommand,
+} from "@stedi/sdk-client-stash";
+import { GetGuideCommand } from "@stedi/sdk-client-guides";
+
+const buckets = mockBucketClient();
+const translate = mockTranslateClient();
+const stash = mockStashClient();
+const guides = mockGuideClient();
+
+const sample855 = fs.readFileSync(
+  "./src/resources/X12/5010/855/inbound.edi",
+  "utf8"
+);
+
+test.before(() => {
+  nock.disableNetConnect();
+  mockExecutionTracking(buckets);
+});
+
+test.after.always(() => {
+  buckets.reset();
+  translate.reset();
+  stash.reset();
+});
+
+test("translates, delivers & acks an incoming X12 file", async (t) => {
+  // loading incoming EDI file from S3
+  buckets.on(GetObjectCommand, {}).resolves({
+    body: sdkStreamMixin(Readable.from([new TextEncoder().encode(sample855)])),
+  });
+
+  // mock translate response
+  translate.on(TranslateX12ToJsonCommand).resolvesOnce({
+    output: { transactionSets: [{}], envelope: { controlNumber: "123456" } },
+  });
+
+  stash
+    .on(GetValueCommand, { key: "lookup|ISA|14/ANOTHERMERCH" }) // mock sending partner lookup
+    .resolvesOnce({ value: { partnerId: "another-merchant" } })
+    .on(GetValueCommand, { key: "lookup|ISA|ZZ/THISISME" }) // mock receiving partner lookup
+    .resolvesOnce({ value: { partnerId: "this-is-me" } })
+    .on(GetValueCommand, { key: "partnership|another-merchant|this-is-me" }) // mock partnership lookup
+    .resolvesOnce({
+      value: {
+        transactionSets: [
+          {
+            acknowledgmentConfig: {
+              acknowledgmentType: "997",
+            },
+            description:
+              "Purchase Order Acknowledgements received from ANOTHERMERCH",
+            destinations: [
+              {
+                destination: {
+                  type: "webhook",
+                  url: "https://webhook.site/TESTING",
+                  verb: "POST",
+                },
+              },
+            ],
+            guideId: "01GRRSB3TXW9Z4J2J75Z9RPR3S",
+            receivingPartnerId: "this-is-me",
+            sendingPartnerId: "another-merchant",
+            usageIndicatorCode: "T",
+          },
+          {
+            description: "Outbound 997 Acknowledgments",
+            destinations: [
+              {
+                destination: {
+                  bucketName: "ab33fa45-c67a-4653-bf5a-0bb85a90b332-sftp",
```

Comment:
nit: not sure if you want to swap out the bucket name (and guide id above) to something more generic? guessing these are real values from your test account

Reply:
Good catch, addressed.